beginner_source/Pretraining_Vgg_from_scratch.py

```diff
@@ -0,0 +1,659 @@
+    print(f'top 1 val acc  %: {top_1_acc}')
+    print(f'top 5 val acc  %: {top_5_acc}')
+
+
+    print(f'top 1 train acc : {train_acc[0]}  top 5 train acc : {train_acc[1]}')
+    print(f'train_size :{train_num}')
+    top_1_train ,top_5_train   = 100*train_acc[0]/train_num, 100*train_acc[1]/train_num
+    print(f'top 1 train acc  %: {top_1_train}')
+    print(f'top 5 train acc  %: {top_5_train}')
+
+
+    scheduler.step(top_5_acc)
+
+
+
+######################################################################
+# (Optional) Additional Exercise: ImageNet
+# --------------------------------------------
+#
+# You can apply the same model that we have trained above with another popular dataset called ImageNet:  
+
+class Custom_ImageNet(ImageNet) :
+    def __init__(self,root,transform = None,multi=False,s_max=None,s_min=256,split=None,val=False):
+
+        self.multi = multi
+        self.s_max = 512
+        self.s_min= 256
+        if multi :
+            self.S = np.random.randint(low=self.s_min,high=self.s_max)
+        else :
+            self.S = s_min
+            transform = A.Compose(
+                    [
+                        A.Normalize(),
+                        A.SmallestMaxSize(max_size=self.S),
+                        A.RandomCrop(height =224,width=224),
+                        A.HorizontalFlip()
+                    ]
+
+            )
+        super().__init__(root,transform=transform,split=split)
+        self.val =val
+        self.multi = multi
+    def __getitem__(self, index: int) :
+        """
+        Args:
+            index (int): Index
+
+        Returns:
+            tuple: (image, target) where target is index of the target class.
+        """
+        path, target = self.samples[index]
+        img = self.loader(path)
+        # doing this so that it is consistent with all other datasets
+        # to return a PIL Image
+        img=np.array(img)
+        img = Image.fromarray(img)
+
+        if img.mode == 'L' : img = img.convert('RGB')
+        img=np.array(img,dtype=np.float32)
+
+
+        if self.transform is not None:
+            img = self.transform(image=img)
+            if len(img['image'].shape) == 3 and self.val==False :
+                img = A.RGBShift()(image=img['image'])
+            img = img['image']
+
+        if self.target_transform is not None:
+            target = self.target_transform(target)
+        # print(img)
+        img=img.transpose((2,0,1))
+
+        return img, target
+
+if DatasetName == 'ImageNet' :
+    train_data= Custom_ImageNet(root='ImageNet',split='train')
+    val_data= Custom_ImageNet('ImageNet',split='val',val=True)
+    val_data.val= True
+    val_data.s_min = test_min
+    val_data.transform=    A.Compose(
+                    [
+                        A.Normalize(),
+                        A.SmallestMaxSize(max_size=val_data.S),
+                        A.CenterCrop(height =224,width=224)
+                    ]
+
+                )
```

Comment:
Can you please run `lintrunner -a` (from [here](https://pypi.org/project/lintrunner/)) on the code blocks throughout using the `.lintrunner.toml` config provided in the base PyTorch repo dir? The use of whitespace is inconsistent and hard to read